binding/lua/multiverso.lua

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env lua
+
+mv = {}
+
+ffi = require('ffi')
+ffi.cdef[[
+    typedef void* TableHandler;
+    void MV_Init(int* argc, char* argv[]);
+    void MV_ShutDown();
+    void MV_Barrier();
+    int MV_NumWorkers();
+    int MV_WorkerId();
+    int MV_ServerId();
+
+    // Array Table
+    void MV_NewArrayTable(int size, TableHandler* out);
+    void MV_GetArrayTable(TableHandler handler, float* data, int size);
+    void MV_AddArrayTable(TableHandler handler, float* data, int size);
+]]
+
+package.cpath = "../../build/src/?.so;" .. package.cpath
+libmv_path = package.searchpath('libmultiverso', package.cpath, '')
+libmv = ffi.load(libmv_path)
+
+function mv.init(args)
+    args = args or {}
+    argc = ffi.new("int[1]", #args)
+    argv = ffi.new("char*[?]", #args)
+    for i = 1, #args do
+        argv[i - 1] = ffi.new("char[1]")
+        ffi.copy(argv[i - 1], args[i])
+    end
+    libmv.MV_Init(argc, argv)
+end
+
+function mv.barrier()
+    libmv.MV_Barrier()
+end
+
+function mv.shutdown()
+    libmv.MV_ShutDown()
+end
+
+function mv.num_workers()
+    return libmv.MV_NumWorkers()
+end
+
+function mv.worker_id()
+    return libmv.MV_WorkerId()
+end
+
+function mv.server_id()
+    return libmv.MV_ServerId()
+end
+
+mv.ArrayTableHandler = {}
+
+function mv.ArrayTableHandler:new(size)
+    tbh = {}
+    size = size or 0
+    setmetatable(tbh, self)
+    self.__index = self
```

Comment:
I think both are OK, the reference I use is [1].

From my limited lua knowledge, if we set the `__index` to base class, `self.__index = self` will be done via `setmetatable()`, so it's just a different expression?

Or you have other concern like efficiency? Maybe I can write some simple test for evaluation.

[1] https://www.lua.org/pil/16.2.html